mempool.go

```diff
@@ -2026,11 +2026,13 @@ func (mp *txMemPool) ProcessTransaction(tx *dcrutil.Tx, allowOrphan,
 
 	// Potentially accept the transaction to the memory pool.
 	var isOrphan bool
-	_, err := mp.maybeAcceptTransaction(tx, true, rateLimit)
+	missingParents, err := mp.maybeAcceptTransaction(tx, true, rateLimit)
 	if err != nil {
 		return err
 	}
 
+	isOrphan = len(missingParents) != 0
```

Comment:
Also, while you're here, please add a return after `mp.processOrphans(tx.Sha())`, remove the `else` clause, and unindent the other block so it looks like the upstream project code.